src/python/ensembl/io/genomio/manifest/manifest.py

```diff
@@ -0,0 +1,176 @@
+
+        Args:
+            manifest_dir: directory where the files are contained.
+        """
+        self.root_dir = manifest_dir
+        self.file_path = manifest_dir / "manifest.json"
+        self.files: dict = {}
+
+    def create(self) -> None:
+        """Creates a manifest file from the files in a directory."""
+        self.get_files_checksums()
+        with self.file_path.open("w") as json_out:
+            json_out.write(json.dumps(self.files, sort_keys=True, indent=4))
+
+    def get_files_checksums(self) -> ManifestDict:
+        """Records all the files in the directory with their checksum."""
+        manifest_files: ManifestDict = {}
+        for subfile in self.root_dir.iterdir():
+            logging.debug(f"Check file {subfile} ({subfile.stem}, {subfile.suffix})")
+            used_file = False
+            if subfile.is_dir():
+                logging.warning("Can't create manifest for subdirectory")
+                continue
+
+            # Delete and skip empty files
+            if subfile.stat().st_size == 0:
+                logging.warning(f"Skip and delete empty file: {subfile}")
+                subfile.unlink()
+                continue
+
+            for name, standard_name in self.names.items():
+                # Either the last element of the stem or the suffix is a known name
+                if subfile.stem.endswith(name) or subfile.suffix == f".{name}":
+                    logging.debug(f"Matched to {name} ({standard_name}) = {subfile}")
+                    used_file = True
+                    md5 = self._get_md5sum(subfile)
+                    file_obj = {"file": subfile.name, "md5sum": md5}
+
+                    # Multiple files stored, each with a name
+                    if standard_name in self.multi_files:
+                        manifest_files.setdefault(standard_name, {})
+                        obj_name = self._prepare_object_name(subfile, name, manifest_files[standard_name])
+                        manifest_files[standard_name][obj_name] = file_obj
+
+                    # Single file/init
+                    else:
+                        manifest_files[standard_name] = file_obj
+
+            if not used_file:
+                logging.warning(f"File {subfile} was not included in the manifest")
+
+        self.files = manifest_files
+        return self.files
+
+    def _prepare_object_name(self, subfile: Path, name: str, manifest_dict: dict[str, dict[str, str]]) -> str:
+        # Prepare object name
+        obj_name = "file"
+        try:
+            # If we recognize the suffix, then the name is the part after the last "_"
+            if subfile.suffix == f".{name}":
+                obj_name = subfile.stem.split(sep="_")[-1]
+            # If we recognize the end of the name, then the name is the part before the last "_"
+            else:
+                obj_name = subfile.stem.split(sep="_")[-2]
+        except IndexError:
+            pass
```

Comment:
```suggestion
        try:
            # If we recognize the suffix, then the name is the part after the last "_"
            if subfile.suffix == f".{name}":
                obj_name = subfile.stem.split(sep="_")[-1]
            # If we recognize the end of the name, then the name is the part before the last "_"
            else:
                obj_name = subfile.stem.split(sep="_")[-2]
        except IndexError:
            obj_name = "file"
```
It is more readable if the fail-safe is placed inside the exception handler.